jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -84,7 +84,9 @@ def process(self, input):
                 gls_opt_model, 'fitoptgls', output_file=self.opt_name
             )
 
-        out_model.meta.cal_step.ramp_fit = 'COMPLETE'
+        if out_model is not None:
+            out_model.meta.cal_step.ramp_fit = 'COMPLETE'
+
```

Comment:
`out_model` is also referenced in line 94. Does that need updating too, to check for None?

Reply:
1) " Does the pipeline handle the returned None gracefully?"
Currently the pipeline does not handle this case gracefully. In the calwebb_detector1 pipeline (the only pipeline that has a step beyond ramp_fit) , the output of the ramp_fit step is intended to be the input model to the gain_scale step, which is the final step of the pipeline.  Like other steps, gain_scale will try to open the ramp_fit output without checking for None:
"with datamodels.open(input) as input_model:". 
This does *not* fail if 'input' is None.

The gain_scale step will then attempt to get the gain_factor from the input model; if that is not successful, the step tries getting gain_factor from the gain reference file (which itself is derived from the input model) . This last try is currently fatal if 'input' is None:

ERROR RUNNING STEP 'Detector1Pipeline':
    Missing 'META.INSTRUMENT.NAME' keyword in header for determining
    instrument.
Traceback (most recent call last):
  File "/internal/1/sc2/git/jwst/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)

For the other pipelines having ramp_fit as a step, it is the final step in the pipeline. So I think there could be 2 approaches  to a fix - either modify ramp fit so that if its output is None pipeline execution is halted, or modify the stpipe machinery to check for this situation.

2) Yes - the out_model code should be checked also.